token-service/src/main/java/org/p2p/token/service/model/TokenServiceNetwork.kt

```diff
@@ -1,6 +1,13 @@
 package org.p2p.token.service.model
 
-enum class TokenServiceNetwork {
-    SOLANA,
-    ETHEREUM
+enum class TokenServiceNetwork(val networkName: String) {
+    SOLANA("solana"),
+    ETHEREUM("ethereum");
+
+    companion object {
+        fun getValueOf(stringValue: String): TokenServiceNetwork {
+            return TokenServiceNetwork.values().firstOrNull { it.networkName == stringValue }
```

Comment:
I would add case-insensitive comparison just in case
```suggestion
            TokenServiceNetwork.values().firstOrNull { it.networkName.equals(stringValue, ignoreCase = true) }
```